server/routers/viewer.tsx

```diff
@@ -78,4 +73,52 @@ export const viewerRouter = createRouter()
 
       return bookings;
     },
+  })
+  .mutation("updateProfile", {
+    input: z.object({
+      username: z.string().optional(),
+      name: z.string().optional(),
+      description: z.string().optional(),
+      avatar: z.string().optional(),
+      timeZone: z.string().optional(),
+      weekStart: z.string().optional(),
+      hideBranding: z.boolean().optional(),
+      theme: z.string().optional(),
+      completedOnboarding: z.boolean().optional(),
+      locale: z.string().optional(),
+    }),
+    async resolve({ input, ctx }) {
+      const { user, prisma } = ctx;
+      const { name, avatar, timeZone, weekStart, hideBranding, theme, completedOnboarding, locale } = input;
+
+      const data: Prisma.UserUpdateInput = {
+        name,
+        avatar,
+        timeZone,
+        weekStart,
+        hideBranding,
+        theme,
+        completedOnboarding,
+        locale,
+        bio: input.description,
+      };
```

Comment:
We don't need to construct if they follow the same name
```suggestion
      const data: Prisma.UserUpdateInput = {
        ...input,
      };
```